src/main/java/com/dev/cinema/controllers/UserController.java

```diff
@@ -4,25 +4,28 @@
 import com.dev.cinema.mapper.UserMapper;
 import com.dev.cinema.service.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @RequestMapping("/users")
 public class UserController {
     private final UserService userService;
-    private final UserMapper userMapper;
+    private final UserMapper mapper;
 
     @Autowired
     public UserController(UserService userService, UserMapper userMapper) {
         this.userService = userService;
-        this.userMapper = userMapper;
+        this.mapper = userMapper;
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto getByEmail(@RequestParam String email) {
-        return userMapper.fromUserToUserResponseDto(userService.findByEmail(email).orElseThrow());
+    public UserResponseDto getByEmail(Authentication authentication) {
```

Comment:
should we change this method? Let's leave it as it was

Reply:
Fixed.